pkg/image/vulnerability_scan.go

```diff
@@ -0,0 +1,118 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+
+	shipwrightv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+)
+
+const (
+	VulnerabilityCountLimit = 50 // Number of vulnerabilities to be added to buildrun output.
+)
+
+func RunVulnerabilityScan(ctx context.Context, imagePath string, settings shipwrightv1alpha1.VulnerabilityScanOptions) ([]shipwrightv1alpha1.Vulnerability, error) {
+
+	trivyArgs := []string{"image", "--quiet", "--format", "json", "--input", imagePath}
+
+	if settings.IgnoreOptions != nil {
+		if settings.IgnoreOptions.Severity != "" {
+			severity := getSeverity(settings.IgnoreOptions.Severity)
+			trivyArgs = append(trivyArgs, "--severity", severity)
+		}
+		if len(settings.IgnoreOptions.Issues) > 0 {
+			// Create a file with vulnerabilities to be ignored.
+			ignoreFile, err := os.CreateTemp("", "ignore")
+			if err != nil {
+				log.Printf("Could not create file for ignored vulnerabilities: %v\n", err)
+				return nil, err
+			}
+			defer os.Remove(ignoreFile.Name())
+			for _, vul := range settings.IgnoreOptions.Issues {
+				_, err := ignoreFile.WriteString(vul + "\n")
+				if err != nil {
+					log.Printf("Error writing to ignore file for vulnerabilities: %v\n", err)
+					return nil, err
+				}
+			}
+			ignoreFile.Close()
+			trivyArgs = append(trivyArgs, "--ignorefile", ignoreFile.Name())
+		}
+		trivyArgs = append(trivyArgs, "--ignore-unfixed", strconv.FormatBool(settings.IgnoreOptions.Unfixed))
+	}
+
+	cmd := exec.CommandContext(ctx, "trivy", trivyArgs...)
+
+	// Print the command to be executed
+	log.Println(cmd.String())
+
+	cmd.Stdin = nil
+
+	result, err := cmd.CombinedOutput()
+
+	if err != nil {
+		log.Println("failed to run trivy : ", string(result))
```

Comment:
Nit: Assuming output in result is multiline.
```suggestion
		log.Printf("failed to run trivy:\n%s", string(result))
```

Reply:
Done